lib/src/visitor/async_evaluate.dart

```diff
@@ -140,6 +140,9 @@ class _EvaluateVisitor
   /// All modules that have been loaded and evaluated so far.
   final _modules = <Uri, Module>{};
 
+  /// Configuration ids seen by a module by URI.
+  final _moduleConfigurationIds = <Uri, int>{};
```

Comment:
Rather than explicitly translating configurations to integers, why not just store the `Configuration` object itself as the value here?

Reply:
Upon further testing, indeed the instances are different, I could still store the original Configuration and just compare their IDs later (or actually implement an `==` override), thoughts?

Reply:
I guess it makes sense that object equality isn't sufficient, since we create derivative configurations for `@forward` rules. Let's nail down the spec before we continue here so we at least know the semantics we're trying to fit.

Reply:
Updated PR following suggestions on spec to get an idea how that looks like